pkg/toolkit/persistence.go

```diff
@@ -0,0 +1,41 @@
+// Copyright © 2023 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package toolkit
+
+import (
+	"context"
+
+	"github.com/hyperledger/firefly-common/pkg/fftypes"
+	"github.com/hyperledger/firefly-transaction-manager/pkg/apitypes"
+)
+
+type Persistence interface {
```

Comment:
Would it not make sense to just embed the `persistence.TransactionPersistence` interface into the toolkit, rather than redefining it here?

Reply:
`persistence` is an internal package. This is not a redefinition, it's the definition of transaction persistence, it gets referenced in the internal `persistence` package.

Reply:
@peterbroadhurst 
Here is the current layout:

`pkg/toolkit/persistence.go` only contains transaction-related functions and gets passed into TX handler.
`internal/persistence/persistence.go` contains a complete set of functions which composites the toolkit.Persistence interface

Which I think you are OK with.

Reply:
I agree with passing the implementation of this interface (as defined in `persistence.TransactionPersistence` - per comment above) to the TX Handler on the `toolkit` API set.

However, I think this should be defined in the `persistence` package, alongside all the other persistence interfaces.